content/installation/upgrading-manager.md

```diff
@@ -2,96 +2,113 @@
-      For high availability clusters, you must also run ```https://github.com/cloudify-cosmo/cloudify-dev/blob/master/scripts/delete_cluster_4_0_1.py```.
+To backup agent certificates:
 
-1. (Optional) We recommend that you run this command to remove the profile directory of this Cloudify Manager from your local `~/.cloudify/profiles` directory:
-   ```rm -rf ~/.cloudify/profiles/{{ your Manager’s IP address }}```
+* Backup the `/etc/cloudify/ssl` directory to a remote host.
 
-    Cloudify Manager is now completely removed from the virtual instance.
+## Uninstall Cloudify Manager
 
-1. [Install]({{< relref "installation/installing-manager.md" >}}) a new Cloudify Manager.
+For instructions on how to uninstall the Cloudify Manager 4.3 and above, go to: [Uninstalling Cloudify Manager]({{< relref "installation/installing-manager.md#uninstalling-cloudify-manager" >}})
+
+To uninstall Cloudify Manager 4.1 or above:
+
+* To remove Cloudify Manager, run: ```cfy teardown -f```
+
+To uninstall Cloudify Manager 4.0.x:
+
+1. To download the teardown script, run: ```curl -o ~/cfy_teardown_4_0_0.sh https://raw.githubusercontent.com/cloudify-cosmo/cloudify-dev/master/scripts/cfy_teardown_4_0_0.sh```
+1. To remove Cloudify Manager, run: ```sudo bash cfy_teardown_4_0_0.sh -f```
+1. Apply ```patch-1```.
+1. (For high availability clusters only) To download the script to delete the cluster configuration, run: ```https://github.com/cloudify-cosmo/cloudify-dev/blob/master/scripts/delete_cluster_4_0_1.py```
+1. (For high availability clusters only) To run the script to delete the cluster configuration, run: ```sudo delete_cluster_4_0_1.py```
+1. (Optional) We recommend that you remove the profile directory of this Cloudify Manager from your local `~/.cloudify/profiles` directory: ```rm -rf ~/.cloudify/profiles/{{ your Manager’s IP address }}```
+
+## Install Cloudify Manager
+
+For instructions on how to install the Cloudify Manager, go to: [Installing Cloudify Manager]({{< relref "installation/installing-manager.md" >}})
+
+## Restore Snapshot
+
+To restore the snapshot:
 
 1. To upload the previously created snapshot to the new Manager, run:
-   ```cfy snapshots upload {{ /path/to/the/snapshot/file }} --snapshot-id my_snapshot```<br>
+   ```cfy snapshots upload {{ /path/to/the/snapshot/file }} --snapshot-id my_snapshot```
 
-1. To restore the snapshot to Cloudify 4.x and above, run: <br>
+1. To restore the snapshot to Cloudify 4.x and above, run:
 
   * For Cloudify Manager 4.0.1 and above: ```cfy snapshots restore my_snapshot --restore-certificates```
   * For Cloudify Manager 4.0.0: ```cfy snapshots restore my_snapshot```
 
 1. To check the status of the execution after it is complete, run:
    ```cfy executions list --include-system-workflows```
 
+## Migrate Agents
+
 1. (Optional) To apply the agent certificates from the previous Manager, run these commands to replace the new SSL directory with the copied one.
```

Comment:
This must NOT be done unless the certificates are referring to the same IPs/hostnames that are relevant for the new manager as well as the old one. Otherwise the agents simply can't work.
These steps should probably be changed to include taking a copy of the SSL certs dir in case someone doesn't know exactly what they're doing but blindly follows them.

Also, this agent cert restore should only be required on an in-place upgrade from 4.0.0

Reply:
I'll add: (For in-place upgrade from Cloudify Manager 4.0.0)
Where is the ssl certs dir?

Reply:
/etc/cloudify/ssl- it's the one referenced in the commands